js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,384 @@
+          )}
+        </div>
+      </div>,
+    ]);
+
+    return items;
+  }
+
+  content() {
+    const settings = app.extensionData.getSettings(this.extension.id);
+
+    return (
+      <div className="ExtensionPage-settings">
+        <div className="container">
+          {typeof app.extensionData[this.extension.id] === 'function' ? (
+            <Button onclick={app.extensionData[this.extension.id].bind(this)} className="Button Button--primary">
+              {app.translator.trans('core.admin.extension.open_modal')}
+            </Button>
+          ) : settings ? (
+            <div className="Form">
+              {this.getSettings(settings)}
+              <div className="Form-group">{this.submitButton()}</div>
+            </div>
+          ) : (
+            <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.no_settings')}</h2>
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  topItems() {
+    const items = new ItemList();
+
+    items.add('version', <span className="ExtensionVersion">{this.extension.version}</span>);
+
+    if (!this.isEnabled()) {
+      const uninstall = () => {
+        if (confirm(app.translator.trans('core.admin.extension.confirm_uninstall'))) {
+          app
+            .request({
+              url: app.forum.attribute('apiUrl') + '/extensions/' + this.extension.id,
+              method: 'DELETE',
+            })
+            .then(() => window.location.reload());
+
+          app.modal.show(LoadingModal);
+        }
+      };
+
+      items.add(
+        'uninstall',
+        <Button icon="fas fa-trash-alt" className="Button Button--primary" onclick={uninstall.bind(this)}>
+          {app.translator.trans('core.admin.extension.uninstall_button')}
+        </Button>
+      );
+    }
+
+    return items;
+  }
+
+  infoItems() {
+    const items = new ItemList();
+
+    if (this.extension.authors) {
+      let authors = [];
+
+      Object.keys(this.extension.authors).map((author, i) => {
+        const link = this.extension.authors[author].homepage
+          ? this.extension.authors[author].homepage
+          : 'mailto:' + this.extension.authors[author].email;
+
+        authors.push(
+          <Link href={link} external={true} target="_blank">
+            {this.extension.authors[author].name}
+          </Link>
+        );
+      });
+
+      items.add('authors', [icon('fas fa-user'), <span>{punctuateSeries(authors)}</span>]);
+    }
+
+    const infoData = {};
+
+    if (this.extension.source || this.extension.support) {
+      infoData.source = {
+        icon: 'fas fa-code',
+        href: this.extension.source ? this.extension.source.url : this.extension.support.source,
+      };
+    }
+
+    Object.keys(this.infoFields).map((field) => {
+      if (this.extension.extra['flarum-extension'].info && this.extension.extra['flarum-extension'].info[field]) {
```

Comment:
Set `this.extension.extra['flarum-extension']` to a temporary variable within the map function for readability.